R/Profile.R

```diff
@@ -0,0 +1,80 @@
+#
+# Copyright (c) TileDB Inc. under the MIT License
+#
+
+#' An S4 class for a TileDB Profile object
+#'
+#' @slot ptr An external pointer to the underlying implementation
+#' @exportClass tiledb_profile
+setClass("tiledb_profile",
+  slots = list(ptr = "externalptr")
+)
+
+setMethod(
+  "raw_dump",
+  signature(object = "tiledb_profile"),
+  definition = function(object) libtiledb_profile_dump(object@ptr)
+)
+
+#' Create a 'tiledb_profile' object
+#'
+#' @param name (optional) Name for the profile.
+#' @param dir (optional) Directory to create the profile in.
+#' @return 'tiledb_profile' object
+#' @export tiledb_profile
+tiledb_profile <- function(name = NULL, dir = NULL) {
+  stopifnot(`The 'name' for the profile must be null or a character type` = is.null(name) || is.character(name))
+  stopifnot(`The 'dir' for the profile must be null or a character type` = is.null(name) || is.character(name))
+  ptr <- libtiledb_profile_new(name, dir)
+  profile <- new("tiledb_profile", ptr = ptr)
+  invisible(profile)
+}
+
+
+tiledb_profile_load <- function(name = NULL, dir = NULL) {
+  stopifnot(`The 'name' for the profile must be null or a character type` = is.null(name) || is.character(name))
+  stopifnot(`The 'dir' for the profile must be null or a character type` = is.null(name) || is.character(name))
+  ptr <- libtiledb_profile_load(name, dir)
+  profile <- new("tiledb_profile", ptr = ptr)
+  invisible(profile)
+}
```

Comment:
Is there a reason we're invisibly returning the profiles created/loaded by these functions? I would think we'd want these to be visible
```suggestion
tiledb_profile <- function(name = NULL, dir = NULL) {
  stopifnot(`The 'name' for the profile must be null or a character type` = is.null(name) || is.character(name))
  stopifnot(`The 'dir' for the profile must be null or a character type` = is.null(name) || is.character(name))
  ptr <- libtiledb_profile_new(name, dir)
  profile <- new("tiledb_profile", ptr = ptr)
  return(profile)
}


tiledb_profile_load <- function(name = NULL, dir = NULL) {
  stopifnot(`The 'name' for the profile must be null or a character type` = is.null(name) || is.character(name))
  stopifnot(`The 'dir' for the profile must be null or a character type` = is.null(name) || is.character(name))
  ptr <- libtiledb_profile_load(name, dir)
  profile <- new("tiledb_profile", ptr = ptr)
  return(profile)
}
```